### PR TITLE
Fix ProtocolLib message listener and refactor Shear & Regrow spells

### PR DESCRIPTION
**Pull request by JasperLorelai**
- `RegrowSpell` can now target `bogged`.
- `ShearSpell` can now target all [`Shearable`](https://jasperlorelai.eu/paperdocs/io/papermc/paper/entity/class-use/Shearable.html#org.bukkit.entity). Added `vanilla-sheep-shearing` boolean option for backward compatibility, which defaults to `false`.
- Fixed an issue with the `DisguiseSpell` sending errors when LibsDisguises wasn't present.

### CODE REVIEW

**Comment by JasperLorelai** [comment]
aa8336e requires comment changes due to merging of https://github.com/PaperMC/Paper/pull/11715/

**Comment by JasperLorelai** [comment]
Removed removal suppressions on aa8336e.

File: core/src/main/java/com/nisovin/magicspells/Spell.java [hunk 1]
```diff
@@ -911,15 +911,14 @@ protected <T extends Keyed> ConfigData<T> getConfigDataRegistryEntry(String key,
 	}
 
 	/**
-	 * @param key Path for the keys to be read from. If the path is set to something like "filter", the keys will
-	 *            be read from spell config section under the "filter" section.
+	 * @param key Path for the String or Config SpellFilter to be read from.
```
Comment (TonytheMacaroni) [inline core/src/main/java/com/nisovin/magicspells/Spell.java:914]: Capitalizing string and config here is a bit odd. Also, aren't they referred to as "section-based" spell filters? All spell filters are in configs, so calling them config spell filters is a bit weird.

File: core/src/main/java/com/nisovin/magicspells/spells/buff/ext/DisguiseSpell.java [hunk 2]
```diff
@@ -53,18 +52,20 @@ public DisguiseSpell(MagicConfig config, String spellName) {
 		burning = getConfigBoolean("burning", false);
 		glowing = getConfigBoolean("glowing", false);
 
-		if (isConfigSection("disguise")) {
-			ConfigurationSection disguiseSection = getConfigSection("disguise");
-			if (disguiseSection != null) entityData = new EntityData(disguiseSection);
+		if (!isConfigSection("disguise")) return;
```
Comment (TonytheMacaroni) [inline core/src/main/java/com/nisovin/magicspells/spells/buff/ext/DisguiseSpell.java:55]: Missing a newline here.

File: core/src/main/java/com/nisovin/magicspells/spells/targeted/SlimeSizeSpell.java [hunk 3]
```diff
@@ -37,9 +37,9 @@ public CastResult cast(SpellData data) {
 
 	@Override
 	public CastResult castAtEntity(SpellData data) {
-		if (!(data.target() instanceof Slime target)) return noTarget(data);
+		if (!SLIME.isValidTarget(data.target())) return noTarget(data);
```
Comment (TonytheMacaroni) [inline core/src/main/java/com/nisovin/magicspells/spells/targeted/SlimeSizeSpell.java:40]: Is this change necessary? The instanceof seems more clear.

**Review by TonytheMacaroni — CHANGES_REQUESTED** [review]
Some small nitpicks.

File: core/src/main/java/com/nisovin/magicspells/castmodifiers/conditions/UsingCondition.java [hunk 4]
```diff
@@ -1,47 +1,50 @@
...
 
-	private MagicItemData itemData = null;
-	
+	private MagicItemData itemData;
+
 	@Override
 	public boolean initialize(@NotNull String var) {
 		itemData = MagicItems.getMagicItemDataFromString(var);
-		return itemData != null;
+		return true;
```
Comment (TonytheMacaroni) [inline core/src/main/java/com/nisovin/magicspells/castmodifiers/conditions/UsingCondition.java:22]: This should be `itemData != null || var.isEmpty()`. An invalid magic item should cause an error.

Comment (TonytheMacaroni) [reply core/src/main/java/com/nisovin/magicspells/castmodifiers/conditions/UsingCondition.java:22]: Actually, probably should do:
```java
if (var.isEmpty()) return true;
itemData = MagicItems.getMagicItemDataFromString(var);
return itemData != null;
```

**Pull request merged**